internal/target/target.go

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package target provides functionality to parse target URI strings.
+package target
```

Comment:
Another thing is, it feels like an overkill to put this function in its own package. This is just a utility function. And there probably be another function in this package.
But anyway, this is OK. Up to you.

Reply:
I couldn't find any other package within the internal directory which would be suitable.

Reply:
Leave it in `internal`? Or `internal/utils`? Target is also OK.

Reply:
Hmm ... I'm not a big fan of creating a utils package and putting a bunch of unrelated stuff into it.

I could either leave it as is, or add a `ParseTarget` exported var in the `internal` package and let the functionality be implemented in `resolver_conn_wrapper.go` (as it was earlier) and additionally just have it set `internal.ParseTarget` to its own `parseTarget` in an `init` function.

Will wait to hear from you and Doug about the best way forward.

Reply:
> package and let the functionality be implemented in resolver_conn_wrapper.go (as it was earlier) and additionally just have it set internal.ParseTarget to its own parseTarget in an init function.

The reason to put the implementations elsewhere is for breaking circular dependencies.  Let's not do this kind of thing unless it's needed for that reason.

> it feels like an overkill to put this function in its own package. 

I don't want a package for every function we want to put in `internal`.  Unless we foresee many "target"-related functions, then let's not make this its own package.

- I'm fine with `internal/util` or `internal/grpcutil`.
- I'm marginally OK with putting this straight into `internal`, with the caveat that that package intentionally does not import much because its primary goal is to break circular dependencies.  We ideally would move everything in there for that purpose into a sub-package (that has no grpc imports) in that case.
- Another idea: `internal/resolver`, which already exists (even if it's not a package currently).

Reply:
Went with `internal/grpcutil`.